mayastor/tests/rebuild.rs

```diff
@@ -0,0 +1,1050 @@
+            )
+            .await
+        );
+    }
+}
+
+/// Test rebuild while running front-end I/O.
+/// Note: This test can take some time to complete because it is running fio and
+/// then comparing the contents of the children to make sure they are in-sync.
+#[tokio::test]
+async fn rebuild_with_load() {
+    init_tracing();
+    let test = start_infrastructure("rebuild_with_load").await;
+    let nexus_hdl = &mut test.grpc_handle("ms1").await.unwrap();
+    let ms2 = &mut test.grpc_handle("ms2").await.unwrap();
+    let ms3 = &mut test.grpc_handle("ms3").await.unwrap();
+
+    const CHILD_SIZE_MB: u64 = 100;
+
+    // Create a nexus with 1 child.
+    let child1 =
+        bdev_create_and_share(ms2, Some(CHILD_SIZE_MB), Some("disk1".into()))
+            .await;
+    create_nexus(nexus_hdl, vec![child1.clone()]).await;
+
+    // Connect to nexus over NVMf.
+    let nexus_uri = publish_nexus(nexus_hdl).await;
+    let nexus_tgt = nvmf_connect(nexus_uri.clone());
+
+    // Run fio against nexus.
+    let (s, r) = unbounded::<i32>();
+    let nvmf_tgt = nexus_tgt.clone();
+    std::thread::spawn(move || {
+        if let Err(e) = s.send(common::fio_verify_size(&nvmf_tgt, NEXUS_SIZE)) {
+            tracing::error!("Failed to send fio complete with error {}", e);
+        }
+    });
+
+    // Let fio run for a bit.
+    std::thread::sleep(Duration::from_secs(2));
+
+    // Add a child and rebuild.
+    let child2 =
+        bdev_create_and_share(ms3, Some(CHILD_SIZE_MB), Some("disk2".into()))
+            .await;
+    add_child(nexus_hdl, &child2, true).await;
+
+    // Wait for fio to complete
+    let fio_result = r.recv().unwrap();
+    assert_eq!(fio_result, 0, "Failed to run fio_verify_size");
+
+    // Wait for rebuild to complete.
+    assert!(
+        wait_for_rebuild_completion(nexus_hdl, &child2, Duration::from_secs(1))
+            .await
+    );
+
+    // Disconnect and destroy nexus
+    nvmf_disconnect(nexus_uri);
+    destroy_nexus(nexus_hdl).await;
+
+    // Check children are in-sync.
+    let child1_tgt = nvmf_connect(child1.clone());
+    let child2_tgt = nvmf_connect(child2.clone());
+    common::compare_devices(&child1_tgt, &child2_tgt, CHILD_SIZE_MB, true);
+    nvmf_disconnect(child1);
+    nvmf_disconnect(child2);
+}
+
+/// Build the infrastructure required to run the tests.
+async fn start_infrastructure(test_name: &str) -> ComposeTest {
+    Builder::new()
+        .name(test_name)
+        .network("10.1.0.0/16")
+        .add_container("ms1")
+        .add_container("ms2")
+        .add_container("ms3")
+        .with_clean(true)
+        .with_prune(true)
+        .build()
+        .await
+        .unwrap()
+}
+
+/// Set up the prerequisites for the tests.
+/// Create a nexus on ms1 and create NVMf shares from ms2 & ms3.
+/// The number of children to be added to the nexus is passed in as a parameter.
+async fn setup_test(
+    test: &ComposeTest,
+    num_nexus_children: usize,
```

Comment:
maybe sanity-check num_nexus_children here. What happens if it is > 2?

Reply:
I've added an assert to check for <3 children, as that is all that is currently required for the tests.